fastify.js

```diff
@@ -794,16 +794,12 @@ function fastify (options) {
       // only call isAsyncConstraint once
       if (isAsync === undefined) isAsync = router.isAsyncConstraint()
       if (rewriteUrl) {
-        const originalUrl = req.url
-        const url = rewriteUrl(req)
-        if (originalUrl !== url) {
-          logger.debug({ originalUrl, url }, 'rewrite url')
-          if (typeof url === 'string') {
-            req.url = url
-          } else {
-            const err = new FST_ERR_ROUTE_REWRITE_NOT_STR(req.url, typeof url)
-            req.destroy(err)
-          }
+        const url = rewriteUrl(req, { log: logger })
```

Comment:
Just to share:

I thought that using the `this.log.info()` would be more "fastify style" - but at this point in the code (there is not the fastify server nor the fastify's request object)

Reply:
@Eomm do you think the `{ log }` object should be bound to `this` rather than being the second parameter?